src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -134,21 +134,38 @@ private void EnsureInit()
         {
             if (_initialized == 0)
             {
+                bool hasMicrosoftFeatureFlagSchema = false;
+
                 IConfiguration featureManagementConfigurationSection = _configuration
+                    .GetChildren()
+                    .FirstOrDefault(section =>
+                        string.Equals(
+                            section.Key,
+                            MicrosoftFeatureManagementFields.FeatureManagementSectionName,
+                            StringComparison.OrdinalIgnoreCase));
+
+                if (featureManagementConfigurationSection != null)
+                { 
+                    hasMicrosoftFeatureFlagSchema = true;
+                }
+                else
+                {
+                    featureManagementConfigurationSection = _configuration
                     .GetChildren()
                     .FirstOrDefault(section =>
                         string.Equals(
                             section.Key,
                             ConfigurationFields.FeatureManagementSectionName,
                             StringComparison.OrdinalIgnoreCase));
+                }
 
                 if (featureManagementConfigurationSection == null && RootConfigurationFallbackEnabled)
                 {
                     featureManagementConfigurationSection = _configuration;
-                }
 
-                bool hasMicrosoftFeatureFlagSchema = featureManagementConfigurationSection != null && 
-                    HasMicrosoftFeatureFlagSchema(featureManagementConfigurationSection);
+                    hasMicrosoftFeatureFlagSchema = featureManagementConfigurationSection != null &&
```

Comment:
Do we need to fallback for Microsoft Feature Flag schema? I don't think we would ever have it without the `feature_management` wrappers. But I suppose we could support customers who want to do that? I prefer we don't support it.

Reply:
@jimmyca15 
I am kind of convinced by Ross. I agree that we might never have it without the "feature_management" wrappers.

In the Microsoft Feature Management schema, "feature_management" is a required property. So I think we should not support fallback for it.

Previously, "FeatureFlags" array was under "FeatureManagement" section. With the complexity brought by the RootConfigurationFallBack behavior, we have to use the "EnsureInit" pattern. Now, the Microsoft Feature Management schema is under "feature_management", we can tell which schema is being used in the constructor. We can simplify the code if we don't support the RootConfigurationFallBack for Microsoft Feature Management schema.